pkg/util/utils.go

```diff
@@ -39,44 +41,53 @@ var (
 
 var log = &logger.Log
 
+func truncateLabelHash(data string) string {
+	return Sha1(data)[:common.HashLength]
+}
 func NormalizeLabels(matchLabels *map[string]string) *map[string]string {
 	newLabels := make(map[string]string)
 	for k, v := range *matchLabels {
-		newLabels[NormalizeLabelKey(k)] = NormalizeName(v)
+		newLabels[NormalizeLabelKey(k, truncateLabelHash)] = NormalizeName(v, truncateLabelHash)
 	}
 	return &newLabels
 }
 
-func NormalizeLabelKey(key string) string {
+func NormalizeLabelKey(key string, shaFn func(data string) string) string {
 	if len(key) <= common.MaxTagScopeLength {
 		return key
 	}
 	splitted := strings.Split(key, "/")
 	key = splitted[len(splitted)-1]
-	return normalizeNameByLimit(key, "", common.MaxTagScopeLength)
+	return normalizeNameByLimit(key, "", common.MaxTagScopeLength, shaFn)
 }
 
-func NormalizeName(name string) string {
-	return normalizeNameByLimit(name, "", common.MaxTagValueLength)
+func NormalizeName(name string, shaFn func(data string) string) string {
+	return normalizeNameByLimit(name, "", common.MaxTagValueLength, shaFn)
 }
 
-func normalizeNameByLimit(name string, suffix string, limit int) string {
+func normalizeNameByLimit(name string, suffix string, limit int, hashFn func(data string) string) string {
```

Comment:
we use sha1 function: https://github.com/vmware-tanzu/nsx-operator/blob/main/pkg/nsx/services/securitypolicy/builder.go#L590 for build ruleIDHash, do you think we also need to change it?

Reply:
Sure, I can update together.